x-pack/legacy/plugins/actions/server/builtin_action_types/webhook.test.ts

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { actionType } from './webhook';
+import { validateConfig, validateSecrets, validateParams } from '../lib';
+
+describe('actionType', () => {
+  test('exposes the action as `webhook` on its Id and Name', () => {
+    expect(actionType.id).toEqual('.webhook');
+    expect(actionType.name).toEqual('webhook');
+  });
+});
+
+describe('secrets validation', () => {
+  test('secrets validation succeeds when secrets is valid', () => {
+    const secrets: Record<string, any> = {
+      username: 'bob',
+      password: 'supersecret',
+    };
+    expect(validateSecrets(actionType, secrets)).toEqual(secrets);
+  });
+
+  test('secrets validation fails when secret password is omitted', () => {
```

Comment:
I'm still getting used to how jest and mocha ends up printing test results, and basically I think they append together all the parent `describe()` strings, and then the `test()` string, so the `secrets validation` isn't really needed here.  A lot (all?) of the action tests have this kind of duplication, would be nice to eventually clean up sometime, but not a biggie, and more important for mocha tests which do not provide line numbers, just the appended describe/test string:-)